### PR TITLE
[No QA] Post deploy message in Slack only when deploy succeeds in all platforms

### PR DESCRIPTION
**Pull request by luacmartins**
### Details
- Splits the `postSlackMessage` action in `postSlackMessageOnFailure` and `postSlackMessageOnSuccess`.
- Moves slack comment actions from `postGithubComment` to `postSlackMessageOnSuccess`.

cc @roryabraham @AndrewGable 

### Fixed Issues
$ https://github.com/Expensify/App/issues/7163

### Tests
1. Next time a deploy fails in some (but not all) platforms, verify that the success message is not posted in `#deployer`, `#announce` and `#expensify-open-source`.
2. Verify that the success message is still posted when the deploy succeeds for all platforms.

- [X] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [X] I verified the PR has a small number of commits behind `main`
- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [ ] I clearly indicated the environment tests should be run in (Staging vs Production)
- [X] I wrote testing steps that cover success & fail scenarios (if applicable)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests & verify they passed on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [X] I verified there are no console errors related to changes in this PR
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I added comments when the code was not self explanatory
    - [ ] I put all copy / text shown in the product in all `src/languages/*` files (if applicable)
    - [X] I followed proper naming convention for platform-specific files (if applicable)
    - [X] I followed style guidelines (in [`Styling.md`](https://github.com/Expensify/App/blob/main/STYLING.md)) for all style edits I made
    - [X] I followed the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs))
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I corroborated the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If I created a new component I verified that a similar component doesn't exist in the codebase

#### PR Reviewer Checklist
- [ ] I verified the PR has a small number of commits behind `main`
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the testing environment is mentioned in the test steps
- [ ] I verified testing steps cover success & fail scenarios (if applicable)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors related to changes in this PR
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified comments were added when the code was not self explanatory
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [ ] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [ ] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If a new component is created I verified that a similar component doesn't exist in the codebase

### QA Steps

- [ ] Verify that no errors appear in the JS console

### CODE REVIEW

File: .github/workflows/platformDeploy.yml [hunk 1]
```diff
@@ -371,46 +371,13 @@ jobs:
           GITHUB_TOKEN: ${{ github.token }}
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK }}
 
-  postGithubComment:
```
Comment (luacmartins) [inline .github/workflows/platformDeploy.yml:374]: This diff looks odd, but `postGithubComment` was just moved to the bottom of the file without the Slack announcement steps in it.

**Review by Beamanator — APPROVED** [review]
This looks good to me, but I'll wait for @roryabraham and / or @AndrewGable to review too since they're masters in this area :D

**Comment by luacmartins** [comment]
@roryabraham @AndrewGable let me know if you wanna review this one.

**Review by roryabraham — APPROVED** [review]
LGTM to me too 👍

**Comment by roryabraham** [comment]
Keep an eye on [this deploy](https://github.com/Expensify/App/actions/runs/1966346326) and what happens when it's finished.

**Comment by luacmartins** [comment]
There were no failures in this deploy, so we'll have to keep an eye out for future deploys. That being said, the real success message did get screwed up! I'll investigate!

Screen Shot 2022-03-11 at 9 51 13 AM

**Pull request merged**